src/socket.rs

```diff
@@ -956,6 +958,18 @@ impl Socket {
         self.inner().set_reuse_port(reuse)
     }
 
+    /// Set value for the `SO_BINDTODEVICE` option on this socket.
+    ///
+    /// If a socket is bound to an interface, only packets received from that
+    /// particular interface are processed by the socket. Note that this only
+    /// works for some socket types, particularly AF_INET sockets.
+    ///
+    /// This function is only available on Linux.
+    #[cfg(all(feature = "all", target_os = "linux"))]
+    pub fn set_bind_interface(&self, interface: CString) -> io::Result<()> {
```

Comment:
Since this a Linux (Unix?) only option it should be moved to the `sys/unix.rs` file, for an example of this see the `Socket::set_mark` function: https://github.com/rust-lang/socket2-rs/blob/be524550f6cfb5557637cb56bbba6e62f2ff5a89/src/sys/unix.rs#L686-L698